compiler/src/iree/compiler/DispatchCreation/BubbleUpExpandShapes.cpp

```diff
@@ -38,7 +38,10 @@ struct BubbleUpExpandShapesPass final
   void runOnOperation() override;
 };
 
-// TODO: move this upstream with other tensor bubbling patterns.
+/// Bubbles a `tensor.expand_shape` op through a `tensor.extract_slice` op. This
+/// pattern only gets applied when the `extract_slice` doesn't modify dimensions
+/// that are expanded by the `expand_shape`.
```

Comment:
IIRC, it only works for static shapes? Dynamic shapes could be very tricky, and I think we're not going to support that in the near future. So we want to document that it only works for static shapes.

Reply:
Yeah, only static shapes.


>and I think we're not going to support that in the near future

Meaning that extract_slice will only support static shapes? Is there any more info on this?